service/matching/tasklist/task_list_manager.go

```diff
@@ -491,10 +497,33 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params AddTaskParams)
 	return syncMatch, err
 }
 
-// DispatchTask dispatches a task to a poller. When there are no pollers to pick
-// up the task or if rate limit is exceeded, this method will return error. Task
-// *will not* be persisted to db
+// DispatchTask dispatches a task to a poller on the active side. When there are no pollers to pick
+// up the task or if the rate limit is exceeded, this method will return error. Task
+// *will not* be persisted to db. On the passive side, dispatches the task to the taskCompleter; it will attempt
+// to complete the task if it has already been started.
 func (c *taskListManagerImpl) DispatchTask(ctx context.Context, task *InternalTask) error {
+	// check if this is the active cluster for the domain
+	domainEntry, err := c.domainCache.GetDomainByID(task.Event.TaskInfo.DomainID)
+	if err != nil {
+		return fmt.Errorf("unable to fetch domain from cache: %w", err)
+	}
+
+	if _, err = domainEntry.IsActiveIn(c.clusterMetadata.GetCurrentClusterName()); err == nil {
```

Comment:
Let's also check if returned bool value to determine active vs passive. Current implementation of `IsActiveIn()` always returns an error when it's returning false value but that doesn't have to be the case in the future. 
I checked usages of `IsActiveIn()` on history side which uses the bool and ignores the err.

Reply:
+1